db/00000000000001_initial_schema.up.sql

```diff
@@ -89,34 +89,24 @@ CREATE TABLE tasks(
     time_precision              BIGINT NOT NULL,           -- the duration to which clients are expected to round their report timestamps, in seconds
     tolerable_clock_skew        BIGINT NOT NULL,           -- the maximum acceptable clock skew to allow between client and aggregator, in seconds
     collector_hpke_config       BYTEA,                     -- the HPKE config of the collector (encoded HpkeConfig message)
-    vdaf_verify_key             BYTEA NOT NULL             -- the VDAF verification key (encrypted)
+    vdaf_verify_key             BYTEA NOT NULL,            -- the VDAF verification key (encrypted)
+
+    -- Authentication token used to authenticate messages to/from the other aggregator.
+    -- These columns are NULL if the task was provisioned by taskprov.
+    aggregator_auth_token_type  AUTH_TOKEN_TYPE,    -- the type of the authentication token
+    aggregator_auth_token       BYTEA,              -- encrypted bearer token
+    -- The aggregator_auth_token columns must either both be NULL or both be non-NULL
+    CONSTRAINT aggregator_auth_token_null CHECK ((aggregator_auth_token_type IS NULL) = (aggregator_auth_token IS NULL)),
```

Comment:
WDYT about unit tests that exercise this constraint, i.e. try to insert values that violate it?

This is my main worry with inserting business logic into the database--it's difficult to test for. But I think our test harness can support such a test without too much pain.

Reply:
Good idea. Our definition of `struct Task` doesn't allow constructing a value where the auth token is set but not its type (or vice-versa), but it's easy enough to write SQL `UPDATE` statements that exercise these constraints with illegal mutations to an existing task so I did that.